### PR TITLE
Feature: Add RichCommand for pinning and unpinning items

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Related to #11481

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by yaira2** [comment]
Can we use for the widgets?

**Comment by hecksmosis** [comment]
We have to wait until we can use context for the sidebar and the widgets

File: src/Files.App/Commands/CommandCodes.cs [hunk 1]
```diff
@@ -16,5 +16,9 @@ public enum CommandCodes
 		EmptyRecycleBin,
 		RestoreRecycleBin,
 		RestoreAllRecycleBin,
+
+		// favorites
```
Comment (yaira2) [inline src/Files.App/Commands/CommandCodes.cs:20]: Should it be "Favorites"?
```suggestion
		// Favorites
```

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by hecksmosis** [comment]
I removed the functions from BaseLayout

**Pull request merged**